docs/plugin-architecture.md

```diff
@@ -0,0 +1,307 @@
+# Plugin Architecture Documentation
+
+This document provides a comprehensive overview of the WooCommerce PayPal Payments plugin architecture, explaining its modular design and how the various components work together.
+
+## Overview
+
+The WooCommerce PayPal Payments plugin is built using a modular architecture powered by the [Syde Modularity](https://github.com/inpsyde/modularity) framework. This design provides:
+
+- **Modular Structure**: Each feature is contained within its own module with clear boundaries
+- **Dependency Injection**: PSR-11 container for service management and dependency resolution
+- **Feature Flags**: Dynamic module loading based on environment variables and filters
+- **Extensibility**: Well-defined extension points for customization and enhancement
+- **Maintainability**: Clear separation of concerns and consistent patterns
+
+## Core Components
+
+### Main Plugin File
+
+The plugin initialization begins in `woocommerce-paypal-payments.php`, which:
+- Loads the Composer autoloader
+- Checks for class existence to prevent conflicts
```

Comment:
This sounds a bit confusing if talking about `class_exists( '\WooCommerce\PayPalCommerce\PluginModule' )`. Maybe like this would be more clear:

```suggestion
- Loads the Composer autoloader if needed (e.g. may be already loaded in some tests)
```